misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -57,53 +56,41 @@ sub run {
-      
-    while ( my $line = $mgi_io->getline() ) {
-
-        chomp($line);
-        my ($accession, $chromosome, $position, $start, $end, $strand,$label, 
-            $status, $marker, $marker_type, $feature_type, $synonym_field) = split(/\t/,$line);
-            
-        $position =~ s/^\s+// if ($position);
-
-        my @synonyms = split(/\|/,$synonym_field) if ($synonym_field);
-        
-	
-        my $desc;
-	if ($marker) {
-	    $desc = $marker;
-	}
-        
+  my $input_file = Text::CSV->new({
+                                    sep_char           => "\t",
+				    empty_is_undef     => 1,
+                                    strict => 1,
+                                    auto_diag => 1,
+                                    allow_loose_quotes => 1,
+				   }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+
+  # skip header
+  $input_file->getline($mgi_io);
+  $input_file->column_names([qw(accession chromosome position start end strand label status marker marker_type feature_type synonym_field)] );
+  while ( my $data = $input_file->getline_hr( $mgi_io ) ) {
+        my $accession = $data->{'accession'};
+        my $marker = defined($data->{'marker'}) ? $data->{'marker'} : undef;
         $acc_to_xref{$accession} = $self->add_xref({ acc        => $accession,
-    	                           		             label      => $label,
-    					                             desc       => $desc,
-    					                             source_id  => $source_id,
-    					                             species_id => $species_id,
-                                                                     dbi        => $dbi,
-    					                             info_type  => "MISC"} );
-        if($verbose and !$desc){
+                                                     label      => $data->{'label'},
+                                                     desc       => $marker,
+                                                     source_id  => $source_id,
+                                                     species_id => $species_id,
+                                                     dbi        => $dbi,
+                                                     info_type  => "MISC"} );
+        if($verbose and !$marker){
     	   print "$accession has no description\n";
         }
         $xref_count++;
-            
+        my @synonyms;
         if(defined($acc_to_xref{$accession})){
-           
+            @synonyms = split(/\|/,$data->{'synonym_field'}) if ($data->{'synonym_field'});
             foreach my $syn (@synonyms) {
                 $self->add_synonym($acc_to_xref{$accession}, $syn, $dbi);
                 $syn_count++;
             }
-            
         }
-        
     }
-      
+    $mgi_io->eof or croak "Error parsing file $file: " . $input_file->error_diag();
     $mgi_io->close();
       
     print $xref_count." MGI Description Xrefs added\n" if($verbose);
```

Comment:
you can use a single if to print both. also it's weird to concatenate the count to the double quoted string, it can just be used there... 
`if ($verbose) {`
`  print "$xref_count MGI Description Xrefs added\n";`
`  print "$syn_count synonyms added\n";`
`}`